lib/db.js

```diff
@@ -1114,6 +1112,67 @@ module.exports = (
       })
   }
 
+  // Reduce redis memory usage by not encoding the keys. Store properties
+  // as fixed indices into arrays instead. Takes a session token object as
+  // its argument, returns an array.
+  function packTokensForRedis (tokens) {
+    return Object.keys(tokens).reduce((result, tokenId) => {
+      const unpackedToken = tokens[tokenId]
+
+      // Use `|| 0` to save space, because it's the shortest falsey value we can stringify
+      const packedToken = REDIS_SESSION_TOKEN_PROPERTIES.map(
+        property => unpackedToken[property] || 0
+      )
+
+      const location = unpackedToken.location
+      if (location) {
+        REDIS_SESSION_TOKEN_LOCATION_PROPERTIES.forEach(
+          property => packedToken.push(location[property] || 0)
```

Comment:
IIUC, this will make it hard to add any additional properties to `REDIS_SESSION_TOKEN_PROPERTIES` in the future.  Is it worth trying e.g. a nested array here rather than flattening the sub-properties of `location`, in the interests of future flexibility?

Reply:
Good point, I'll nest `location`.